File: array/easy/stock_buy_and_sell_one_transcation_allowed/code_1.c
int maximumProfit(int prices[], int pricesSize) {
    // code here
    int minPrice = prices[0];
    int maxProfit = 0;
    
    for(int i = 1; i < pricesSize; i++){
        if(prices[i] < minPrice){
            minPrice = prices[i];
        } else {
            int profit = prices[i] - minPrice;
            if(profit > maxProfit){
                maxProfit = profit;
            }
        }
    }
    
    return maxProfit;
}

File: array/medium/stock_buy_and_sell_multiple_transactions_allowed/code_1.py
from typing import List


class Solution:
    def maximumProfit(self, prices) -> int:
        # code here
        n = len(prices)
        profit = 0
        
        for i in range(0, n - 1):
            if prices[i + 1] > prices[i]:
                profit += prices[i + 1] - prices[i]

        return profit